### ISSUE TITLE
Not able to connect via phpMyAdmin on Synology NAS

### ISSUE THREAD

**Issue by evadue** [docker-library/mariadb#310]
I try to run mariadb in Docker on my Synology NAS using this command and it appears to work fine:

`docker run --name mariadb -v /volume1/docker/mariadb:/var/lib/mysql -e MYSQL_ROOT_PASSWORD=xxx -d mariadb:latest`

I also run phpMyAdmin in Docker:
`docker run --name phpmyadmin -d --link mariadb:db -p 84:80 -v /volume1/docker/phpmyadmin/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php phpmyadmin/phpmyadmin`

When I try to log in through phpMyAdmin I get these errors:

![error](https://user-images.githubusercontent.com/32069841/83021244-01f31580-a02a-11ea-95c7-633627d60fb4.png)

And this line in the mariadb log:

`[Warning] Aborted connection 8 to db: 'unconnected' user: 'unauthenticated' host: '172.17.0.5' (This connection closed normally without authentication)`

If I use the MariaDB package from the Synology Package Center and use
`PMA_HOST=IP:PORT` in phpMyAdmin it works perfectly.

But I would like to also have the db in Docker if possible.

**Comment by wglambert** [comment]
Looks like a duplicate of https://github.com/docker-library/mariadb/issues/264

**Comment by timfosse** [comment]
> Looks like a duplicate of #264

I tried adding `--default-authentication-plugin=mysql_native_password` to my command, but nothing happened, beside a line in the log that says it's a feature of mysql that hasn't been implemented yet.

**Comment by wglambert** [comment]
I didn't have any issues using the `docker-compose.yml` in that thread https://github.com/docker-library/mariadb/issues/264#issuecomment-546668998

![image](https://user-images.githubusercontent.com/26428991/83033906-8757d500-9fec-11ea-8443-fa5e8329cc95.png)

**Comment by timfosse** [comment]
Pretty sure I can't do that on Synology?

**Comment by wglambert** [comment]
Well the issue seems to be something with your host's environment or a configuration issue, and without a practical reproducer there's no method for us to troubleshoot by

You could also try asking over at the [Docker Community Forums](https://forums.docker.com/), [Docker Community Slack](https://dockr.ly/slack), or [Stack Overflow](https://stackoverflow.com/search?tab=newest&q=docker). Since these repos aren't really a user-help forum

**Comment by AbDhops** [comment]
@evadue 
Did you resolved the issue ? How... ?

**Comment by evadue** [comment]
> @evadue
> Did you resolved the issue ? How... ?

No, I gave up. I don't know where to go from here..

**Comment by evadue** [comment]
SOLVED!

After a couple more hours of Googling I came across [this thread](https://www.synoforum.com/threads/phpmyadmin-mariadb-discourse.2783/page-2).

What I had to do was to create a file called `config.inc.php` and map it up with the container:

`-v /volume1/docker/phpmyadmin/config.inc.php:/etc/phpmyadmin/config.inc.php`

And add the following content to the file:

`<?php

require('/etc/phpmyadmin/config.secret.inc.php');

/* Ensure we got the environment */
$vars = array(
    'PMA_ARBITRARY',
    'PMA_HOST',
    'PMA_HOSTS',
    'PMA_VERBOSE',
    'PMA_VERBOSES',
    'PMA_PORT',
    'PMA_PORTS',
    'PMA_USER',
    'PMA_PASSWORD',
    'PMA_ABSOLUTE_URI'
);
foreach ($vars as $var) {
    $env = getenv($var);
    if (!isset($_ENV[$var]) && $env !== false) {
        $_ENV[$var] = $env;
    }
}

/* Arbitrary server connection */
if (isset($_ENV['PMA_ARBITRARY']) && $_ENV['PMA_ARBITRARY'] === '1') {
    $cfg['AllowArbitraryServer'] = true;
}

/* Play nice behind reverse proxys */
if (isset($_ENV['PMA_ABSOLUTE_URI'])) {
    $cfg['PmaAbsoluteUri'] = trim($_ENV['PMA_ABSOLUTE_URI']);
}

/* Figure out hosts */

/* Fallback to default linked */
$hosts = array('db');

/* Set by environment */
if (!empty($_ENV['PMA_HOST'])) {
    $hosts = array($_ENV['PMA_HOST']);
    $verbose = array($_ENV['PMA_VERBOSE']);
    $ports = array($_ENV['PMA_PORT']);
} elseif (!empty($_ENV['PMA_HOSTS'])) {
    $hosts = explode(',', $_ENV['PMA_HOSTS']);
    $verbose = explode(',', $_ENV['PMA_VERBOSES']);
    $ports = explode(',', $_ENV['PMA_PORTS']);
}

/* Server settings */
for ($i = 1; isset($hosts[$i - 1]); $i++) {
    $cfg['Servers'][$i]['host'] = $hosts[$i - 1];
    if (isset($verbose[$i - 1])) {
        $cfg['Servers'][$i]['verbose'] = $verbose[$i - 1];
    }
    if (isset($ports[$i - 1])) {
        $cfg['Servers'][$i]['port'] = $ports[$i - 1];
    }
    if (isset($_ENV['PMA_USER'])) {
        $cfg['Servers'][$i]['auth_type'] = 'config';
        $cfg['Servers'][$i]['user'] = $_ENV['PMA_USER'];
        $cfg['Servers'][$i]['password'] = isset($_ENV['PMA_PASSWORD']) ? $_ENV['PMA_PASSWORD'] : '';
    } else {
        $cfg['Servers'][$i]['auth_type'] = 'cookie';
    }
    $cfg['Servers'][$i]['compress'] = false;
    $cfg['Servers'][$i]['AllowNoPassword'] = true;
}
/*
 * Revert back to last configured server to make
 * it easier in config.user.inc.php
 */
$i--;

/* Uploads setup */
$cfg['UploadDir'] = '';
$cfg['SaveDir'] = '';

/* Include User Defined Settings Hook */
if (file_exists('/etc/phpmyadmin/config.user.inc.php')) {
    include('/etc/phpmyadmin/config.user.inc.php');
}`

And it just worked like that!

**Comment by AbDhops** [comment]
Wowwwwwwwwww... Will try and let you know...

**Comment by AbDhops** [comment]
Well, I tried exact steps as you mentioned. However, issue is still same. See below my docker compose file. Do you see any error ?

```
version: "3.7"

# to Enter in docker 
# docker exec -it MariaDB bash
# docker logs mariadb
services:
  db:
    image: mariadb
    container_name: MariaDB
    restart: always
    environment:
    - MYSQL_ROOT_PASSWORD=test
    volumes:
    - './database:/var/lib/mysql'
    - /etc/localtime:/etc/localtime:ro
       
# docker exec -it MyPhpAdmin bash
  myphpadmin:
    image: phpmyadmin/phpmyadmin:latest
    container_name: MyPhpAdmin
    restart: always
    hostname: phpadmin.test.com
    depends_on:
    - db
    volumes:
    - /etc/localtime:/etc/localtime:ro
    - './phpConfig/config.inc.php:/etc/phpmyadmin/config.inc.php'
    environment:
    - PMA_HOST=db
    - PMA_PORT=3306
    ports:
    - 8086:80
 
```

**Comment by ludipq** [comment]
Try adding

`--link db:db`

And remove `PMA_HOST` and `PMA_PORT`

**Comment by AbDhops** [comment]
I imagine your `config.inc.php ` is the one from [comment](https://github.com/docker-library/mariadb/issues/310#issuecomment-651352471) ?
Unfortunately error still continues...

**Comment by ludipq** [comment]
> I imagine your `config.inc.php ` is the one from [comment](https://github.com/docker-library/mariadb/issues/310#issuecomment-651352471) ?
> Unfortunately error still continues...

Yep.

The only thing I can think of that you can try is adding back in the `PMA_HOST` and `PMA_PORT` and try using [the local Docker IP of the database container](https://stackoverflow.com/questions/17157721/how-to-get-a-docker-containers-ip-address-from-the-host).

If that doesn't work either, I suggest you open your own issue and reference this issue so that people with more experience can help you. I think your problem isn't the same as mine was seeing as adding the config file didn't work for you :/

**Comment by AbDhops** [comment]
So **RTFM**... I learned it in hard way... 
I overlooked the clear warning on MariaDB docker page..

> ### No connections until MySQL init completes
> If there is no database initialized when the container starts, then a default database will be created. While this is the expected behavior, this means that it will not accept incoming connections until such initialization completes. This may cause issues when using automation tools, such as `docker-compose`, which start several containers simultaneously.

For fresh instant, it took almost 11 minutes, until Phpmyadmin (or for that matter any one) to eastablish connection with database.

**Comment by Gofoodi** [comment]
Combination of both topics solved the problem for me. THANK YOU. Searched for the error for several hours now...

> > Looks like a duplicate of #264
> 
> I tried adding `--default-authentication-plugin=mysql_native_password` to my command, but nothing happened, beside a line in the log that says it's a feature of mysql that hasn't been implemented yet.
> 
> Other than that, I can't see any other differences between my command and the one that resolved the problem in #264 ?



> SOLVED!
> 
> After a couple more hours of Googling I came across [this thread](https://www.synoforum.com/threads/phpmyadmin-mariadb-discourse.2783/page-2).
> 
> What I had to do was to create a file called `config.inc.php` and map it up with the container:
> 
> `-v /volume1/docker/phpmyadmin/config.inc.php:/etc/phpmyadmin/config.inc.php`
> 
> And add [the following content ](https://pastebin.com/x7eHgx7K)to the file
> 
> And it just worked like that!
